HAWQ-1473 - document ranger plugin service high availability

lisakowen opened the pull request:
hawq/ranger HA info, including:
- adding subnav entry for HA
- add option step to configure standby RPS
- document new hawq_rps_check_local_interval guc
- update cli and ambari "activating standby" procedures

ictmalili commented on markdown/admin/MasterMirroring.html.md.erb:
```diff
@@ -103,6 +103,8 @@ Upon activation of the standby master, HAWQ reconstructs the state of the master
+	
+6. If you have enabled HAWQ Ranger Authentication in high availability mode in your cluster, you must manually update the Ranger HAWQ service definition to identify the new master node. Refer to [Failover to HAWQ Standby Master](../ranger/ranger-ha.html#rps_ha_cfg_masterfailover) in the HAWQ Ranger documentation for additional information.
 
 6. Query the `gp_segment_configuration` table to verify that segments have registered themselves to the new master:
```
The index number for next item should be incremented.

ictmalili commented on markdown/ranger/ranger-integration-config.html.md.erb:
```diff
@@ -129,6 +129,62 @@ Once the connection between HAWQ and Ranger is configured, you may choose to set
+
+- Installed or upgraded to a version of HAWQ that includes support for HAWQ Ranger Authentication.
+
+- Configured Ranger Admin for high availability.
```
Actually Ranger HA and RPS HA are two separated and independent parts. If the user does not configure Ranger HA, he can also configure RPS HA in HAWQ standby master.

ictmalili commented on markdown/ranger/ranger-integration-config.html.md.erb:
```diff
@@ -129,6 +129,62 @@ Once the connection between HAWQ and Ranger is configured, you may choose to set
+
+- Registered the HAWQ Ranger Plug-in Service on your HAWQ master node.
+
+    The HAWQ Ranger Plug-in Service runs on the HAWQ master node. If you have not yet enabled the Ranger Plug-in Service, refer to [Install Ranger Connectivity to HAWQ](ranger-integration-config.html#jar) for registration instructions. Make sure to identify the Ranger Admin HA proxy when you enable the plug-in.
```
This is not required.

ictmalili commented on markdown/ranger/ranger-integration-config.html.md.erb:
```diff
@@ -129,6 +129,62 @@ Once the connection between HAWQ and Ranger is configured, you may choose to set
+### Configuring the Standby Ranger Plug-in Service <a id="rps_ha_cfg_standbyrps"></a>
+
+The standby Ranger Plug-in Service runs on the HAWQ standby master node, utilizing the same port number as that when running on the master node. To enable HAWQ Ranger high availability, you must register the standby Ranger Plug-in Service on the standby master node, and then restart the standby.
+
```
I think it's not "restart standby". 
If there was only a HAWQ master configured with Ranger, the user wants to configure standby master, he only needs to do "hawq init standby --standby-host=$standbyHost". 

If there was a HAWQ cluster with standby master, the user wants to enable Ranger on whole HAWQ cluster, he needs run the script "enable-ranger-plugin.sh" on both master and standby master, and then restart HAWQ cluster.

dyozie commented on markdown/ranger/ranger-integration-config.html.md.erb:
```diff
@@ -129,6 +129,64 @@ Once the connection between HAWQ and Ranger is configured, you may choose to set
+
+The HAWQ Ranger Plug-in Service runs on the HAWQ master node. If this service goes down, all HAWQ database operations will fail. Configure a highly available HAWQ Ranger Plug-in Service to eliminate possible downtime should this situation occur.
+
+Ranger Admin high availability and HAWQ Ranger Plug-in Service high availability are independent; you can configure HAWQ Ranger Plug-in Service HA without configuring HA for Ranger Admin.
```
Might want to add that configuring HA for both is advised?  Also, this section uses the abbreviated term "Ranger Admin" throughout, which seems confusing to me.  Should probably stick with "Ranger Administration Host" to stay consistent with the first part of the doc.

dyozie commented on markdown/ranger/ranger-integration-config.html.md.erb:
```diff
@@ -129,6 +129,64 @@ Once the connection between HAWQ and Ranger is configured, you may choose to set
+
+Before you configure HAWQ Ranger authentication in high availability mode, ensure that you have:
+
+- Installed or upgraded to a version of HAWQ that includes support for HAWQ Ranger Authentication.
```
Wondering if this first bullet is really needed...

dyozie commented on markdown/ranger/ranger-integration-config.html.md.erb:
```diff
@@ -129,6 +129,64 @@ Once the connection between HAWQ and Ranger is configured, you may choose to set
+    The HAWQ Ranger Plug-in Service runs on the HAWQ master node. If you have not yet enabled the Ranger Plug-in Service, refer to [Install Ranger Connectivity to HAWQ](ranger-integration-config.html#jar) for registration instructions. (Optional) If you have configured Ranger Admin HA, make sure to identify the Ranger Admin HA proxy when you enable the plug-in.
+
+
+### Configuring the Standby Ranger Plug-in Service <a id="rps_ha_cfg_standbyrps"></a>
```
This heading title should just be "Procedure".

dyozie commented on markdown/ranger/ranger-integration-config.html.md.erb:
```diff
@@ -129,6 +129,64 @@ Once the connection between HAWQ and Ranger is configured, you may choose to set
+
+### Configuring the Standby Ranger Plug-in Service <a id="rps_ha_cfg_standbyrps"></a>
+
+The standby Ranger Plug-in Service runs on the HAWQ standby master node, utilizing the same port number as that when the service runs on the master node. To enable HAWQ Ranger high availability, you must register the standby Ranger Plug-in Service on the standby master node, and then restart the standby.
```
This paragraph should be merged with the intro paragraph directly under **Step 3**.  Right now it repeats some of that info.

dyozie commented on markdown/ranger/ranger-integration-config.html.md.erb:
```diff
@@ -129,6 +129,64 @@ Once the connection between HAWQ and Ranger is configured, you may choose to set
+
+The standby Ranger Plug-in Service runs on the HAWQ standby master node, utilizing the same port number as that when the service runs on the master node. To enable HAWQ Ranger high availability, you must register the standby Ranger Plug-in Service on the standby master node, and then restart the standby.
+
+**Note**: If you configured and registered the master HAWQ Ranger Plug-in Service before you initialized your HAWQ standby master node, you do not need to perform the steps in this section.
```
This note is confusing.  The last bullet under prerequisites specifically states that they should have already registered the master plug-in and standby master.  If it's true that setting up the master plug-in first and then configuring the standby master automatically registers the standby plug-in, then this note should probably appear much earlier (before the prerequisites).

dyozie commented on markdown/reference/guc/parameter_definitions.html.md.erb:
```diff
@@ -2147,6 +2149,14 @@ Identifies the port on which the HAWQ Ranger Plug-in Service runs. The `hawq_rps
 
+## <a name="hawq_rps_check_local_interval"></a>hawq\_rps\_check\_local\_interval
+
+When HAWQ Ranger authentication high availability mode is enabled and the Ranger Plug-in Service is active on the standby master node, HAWQ attempts to switch back to the service located on the master node as soon as it becomes available. The HAWQ master periodically attempts to re-establish contact with the service on the local node, using `hawq_rps_check_local_interval` as the polling time inte ...
```
This wording is a bit unclear as to whether it's talking about just the standby ranger service being used (which is the intent) vs whether the entire master node is down and the standby plug-in is being used.  I think it would be better to repeat the wording used in `ranger-ha.html`, which starts "Should the HAWQ master node fail to communicate with the local Ranger Plug-in Service..."

lisakowen commented:
@dyozie - thanks for the review.  i incorporated your edits.  used "Ranger Administration host", as the host was not capitalized in previous uses on the page.  i originally used "Ranger Admin" because that is how it was referred to in some hwx docs.

i moved the note to the prerequisite section.  it seemed out of place in the intro.
